modules/ingestor/src/service/advisory/cve/loader.rs

```diff
@@ -216,4 +276,35 @@ mod test {
 
         Ok(())
     }
+
+    #[test_context(TrustifyContext)]
+    #[test(tokio::test)]
+    async fn divine_purls(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
+        let graph = Graph::new(ctx.db.clone());
+
+        let (cve, digests): (Cve, _) = document("cve/CVE-2024-26308.json").await?;
+
+        let loader = CveLoader::new(&graph);
+        loader
+            .load(("file", "CVE-2024-26308.json"), cve, &digests)
+            .await?;
+
+        let purl = graph
+            .get_package(
+                &Purl::from_str("pkg://maven/org.apache.commons/commons-compress")?,
+                Transactional::None,
+            )
+            .await?;
```

Comment:
Does this affirm that the `purls` field in the response from `/api/v1/vulnerability/CVE-2024-26308` will be non-empty?

Reply:
I dunno TBH. But I hope so.

Reply:
It doesn't:
```
{
  "normative": true,
  "identifier": "CVE-2024-26308",
  "title": "Apache Commons Compress: OutOfMemoryError unpacking broken Pack200 file",
  "description": "Allocation of Resources Without Limits or Throttling vulnerability in Apache Commons Compress.This issue affects Apache Commons Compress: from 1.21 before 1.26.\n\nUsers are recommended to upgrade to version 1.26, which fixes the issue.\n\n",
  "published": "2024-02-19T08:31:50.192Z",
  "modified": "2024-08-02T00:07:19.215Z",
  "withdrawn": null,
  "discovered": null,
  "released": null,
  "cwes": [
    "CWE-770"
  ],
  "average_severity": null,
  "average_score": null,
  "advisories": [
    {
      "uuid": "urn:uuid:2ea5cbf8-2f47-43ca-a234-1fe44a5641d1",
      "identifier": "CVE-2024-26308",
      "hashes": [
        "sha256:a7d8be0b6f068192e56b55a41dff77c8bc06639d60f6213ebd11078316028c37",
        "sha384:3369b48d232b6619ae41fe50022d8d460fe7fc9723ea7af58dba0b66aaafef936ac841a4a9f095806302934c81166395",
        "sha512:8d803e461a66b828ac9fbca0fe4c6fbc9a7413dd0d5a3a00aad53ee148b527a276a6dce79106263dbe1c98695f30bc3227aab3c753c8437f6fa8c6ff3662192b"
      ],
      "issuer": {
        "id": "733e416f-8d89-40c1-b162-609900c49742",
        "name": "apache",
        "cpe_key": null,
        "website": null
      },
      "published": "2024-02-19T08:31:50.192Z",
      "modified": "2024-08-02T00:07:19.215Z",
      "withdrawn": null,
      "title": "Apache Commons Compress: OutOfMemoryError unpacking broken Pack200 file",
      "labels": {
        "type": "cve"
      },
      "severity": null,
      "score": null,
      "cvss3_scores": [],
      "purls": {},
      "sboms": []
    }
  ]
}
```
Unless I'm doing it wrong. I just uploaded the one advisory, and called `/api/v1/vulnerability/CVE-2024-26308`

Reply:
Nevermind. The branch I was testing didn't include your fix. Doh! The purl is there:

```
      "purls": {
        "affected": [
          {
            "base_purl": {
              "uuid": "41997fa4-ddb0-5def-aeba-94e8d7cd9ed6",
              "purl": "pkg://maven/org.apache.commons/commons-compress"
            },
            "version": "[1.21,1.26.0)",
            "context": null
          }
        ]
      },
```

Reply:
Success!